src/orbit_determination/osculating.jl

```diff
@@ -0,0 +1,171 @@
+@doc raw"""
+    OsculatingElements{T <: Number}
+
+Osculating orbital elements of a NEO. 
+
+# Fields 
+
+- `e::T`: eccentricity.
+- `q::T`: perihelion distance [au].
+- `tp::T`: time of pericenter passage [jd]. 
+- `Ω::T`: longitude of ascending node [deg].
+- `ω::T`: argument of pericentre [deg].
+- `i::T`: inclination [deg]. 
+- `a::T`: semimajor axis [au].
+"""
+@auto_hash_equals struct OsculatingElements{T <: Number}
+    e::T
+    q::T
+    tp::T
+    Ω::T
+    ω::T
+    i::T
+    a::T
+    # Inner constructor 
+    function OsculatingElements{T}(e::T, q::T, tp::T, Ω::T, ω::T, i::T, a::T) where {T <: Number}
+        return new{T}(e, q, tp, Ω, ω, i, a)
+    end
+end
+
+# Outer constructors
+function OsculatingElements(e::T, q::T, tp::T, Ω::T, ω::T, i::T, a::T) where {T <: Number}
+    return OsculatingElements{T}(e, q, tp, Ω, ω, i, a)
+end
+
+function OsculatingElements()
+    return OsculatingElements(NaN, NaN, NaN, NaN, NaN, NaN, NaN)
+end
+
+# A OsculatingElements is NaN if all its fields are NaN 
+function isnan(osc::OsculatingElements{T}) where {T <: Number} 
+    return isnan(osc.e) && isnan(osc.q) && isnan(osc.tp) && isnan(osc.Ω) && isnan(osc.ω) && isnan(osc.i) && isnan(osc.a)
+end
+
+# Print method for OsculatingElements
+# Example: 
+# Semimajor axis (a):                 0.8717319220347314 au
+# Eccentricity (e):                   0.4231715487782969
+# Time of pericenter passage (tp):    2021-03-19T03:45:01.293 JDTDB
+# Pericenter distance (q):            0.5028397744678126 au
+# Argument of pericenter (ω):         194.74654283451 deg
+# Inclination (i):                    34.81327005431841 deg
+# Longitude of Ascending Node (Ω):    17.855086873010706 deg
+function show(io::IO, m::OsculatingElements{T}) where {T <: Number} 
+    
+    print(io, rpad("Semimajor axis (a): ", 36), cte(m.a), " au\n")
+    print(io, rpad("Eccentricity (e): ", 36), cte(m.e), "\n")
+    if isnan(m.tp)
+        print(io, rpad("Time of pericenter passage (tp): ", 36), "NaN JDTDB\n")
+    else 
+        print(io, rpad("Time of pericenter passage (tp): ", 36), julian2datetime(cte(m.tp)), " JDTDB\n")
+    end 
+    print(io, rpad("Pericenter distance (q): ", 36), cte(m.q), " au\n")
+    print(io, rpad("Argument of pericenter (ω): ", 36), cte(m.ω), " deg\n")
+    print(io, rpad("Inclination (i): ", 36), cte(m.i), " deg\n")
+    print(io, rpad("Longitude of Ascending Node (Ω): ", 36), cte(m.Ω), " deg\n")
+
+end
+
+@doc raw"""
+    pv2kep(xas, μ = μ_S, jd = JD_J2000)
+
+Compute the orbital elements of the NEO with state vector `xas`. Return a `OsculatingElements` object. 
+
+See also [`eccentricity`](@ref), [`semimajoraxis`](@ref), [`timeperipass`](@ref),
+[`longascnode`](@ref), [`argperi`](@ref) and [`inclination`](@ref).
+
+# Arguments 
+
+- `xas`: State vector of the asteroid `[x, y, z, v_x, v_y, v_z]`. 
+- `μ_S`: Mass parameter of the central body (Sun).
+- `jd`: Orbit epoch of reference in julian days. 
+"""
+function pv2kep(xas, μ = μ_S, jd = JD_J2000)
```

Comment:
What do you think about somehow giving the user the option to return the elements in either ecliptic or equatorial coordinates? Maybe with a `kwarg`? E.g.:
```suggestion
function pv2kep(xas, μ = μ_S, jd = JD_J2000, frame=:equatorial)
```

Reply:
Great idea! I will implement this on future commits.